modules/programs/vscode.nix

```
@@ -212,11 +248,13 @@ in {
             builtins.attrNames (builtins.readDir (ext + "/${subDir}")));
       in if cfg.mutableExtensionsDir then
         mkMerge (concatMap toPaths cfg.extensions)
+        ++ [{ ".vscode/extensions/extensions.json".text = extensionsJson; }]
```

```suggestion
        mkMerge (concatMap toPaths cfg.extensions
        ++ [{ ".vscode/extensions/extensions.json".text = extensionsJson; }])
```

Also I wonder if this prevents installing extensions manually?

---

Why do you suggest this change? I don't think that will work.

Also yes it might prevent installing extensions manually. I'd like some input from someone who uses home manager and VS Code that way and knows what to do about it

---

> I'd like some input from someone who uses home manager and VS Code that way and knows what to do about it

Ask and you shall receive. I'm suggesting these changes because I'm actually testing your code and they're required to make it work.

More in detail: `mkMerge` takes a list of modules and produces a special attrset that can't be further modified, so you want to append to the input list of modules rather than the output.

---

Oh, I misread the suggestion! I'm sorry, I think you're correct. If I accept your suggestion then GitHub will mark this conversation as resolved, so could you make your point about preventing manual extension installation again elsewhere?